src/state/slices/portfolioSlice/portfolioSlice.ts

```diff
@@ -161,8 +165,19 @@ export const accountToPortfolio: AccountToPortfolio = args => {
         portfolio.accounts.byId[accountSpecifier].push(caip19)
         portfolio.accounts.ids.push(accountSpecifier)
 
-        portfolio.assetBalances.byId[caip19] = ethAccount.balance
-        portfolio.assetBalances.ids.push(caip19)
+        // if asset already exist inside assetBalances, add balance to existing balance
+        if (!portfolio.assetBalances.byId[caip19]) {
+          portfolio.assetBalances.byId[caip19] = ethAccount.balance
+        } else {
+          portfolio.assetBalances.byId[caip19] = sumBalance(
+            portfolio.assetBalances.byId[caip19],
+            ethAccount.balance
+          )
+        }
+
+        if (!portfolio.assetBalances.ids.includes(caip19)) {
```

Comment:
we can also do the `Array.from(new Set([...oldArray, newValue]))` pattern here to one line it, but i'm just being stylistic/nit picking here